pages/vote/[id].tsx

```diff
@@ -1,26 +1,71 @@
-import { useRouter } from 'next/router'
-import React from 'react'
-import ViewPoll from '../../components/ViewPoll'
+import React, { useState, useEffect } from "react";
+import { useRouter } from "next/router";
 
+import Button from "@mui/material/Button";
+import { GetServerSideProps } from "next";
+import { resetServerContext } from "react-beautiful-dnd";
+
+import Page from "../../components/Page";
+import VotePoll from "../../components/VotePoll";
+import { GetPollRequest } from '../../components/api';
+import { useFetch } from '../../hooks/useFetch';
 
 function VotePage() {
-  const router = useRouter()
-  const { id } = router.query
-  // useEffect, get the voting question and candidates by id
-  const question = "What Movie Should We Watch?"
-  const pollCandidates = [
-	  "Superman",
-	  "Spiderman",
-	  "Batman",
-	  "Avengers",
-	  "Joker",
-	  "Captain Marvel"
-  ]
+  const router = useRouter();
+
+  const [pollData, getPollData] = useFetch(GetPollRequest);
+
+  useEffect(() => {
+    if (!router.isReady) {
+      return;
+    }
+
+    const { id } = router.query;
+    getPollData(String(id));
+  }, [router.isReady]);
 
+  const SidebarButton = (props: any) => (
+    <Button {...props} style={{ marginBottom: "0.5rem" }}>
+      {props.children}
+    </Button>
+  );
 
+  const Sidebar = (
+    <div
+      style={{
+        display: "flex",
+        flexDirection: "column",
+        minWidth: "8rem",
+        minHeight: "2rem",
+        paddingTop: "5rem",
+      }}
+    >
+      <SidebarButton variant="contained" color="primary">
+        Share
+      </SidebarButton>
+      <SidebarButton variant="outlined">Edit Poll</SidebarButton>
+      <SidebarButton variant="outlined">Close Poll</SidebarButton>
+    </div>
+  );
+
+  const [pageAlert, setPageAlert] = useState<AlertShape | null>(null);
+
+  const result = pollData.data ? (
```

Comment:
This sub-rendering pattern is frowned upon, I would say you should put the `result` code just inside down below and to follow patterns you could put `const Sidebar` in its own file to be its own component or you could do this as a sub-component where its just a const at the top level of this file (same level as` function VotePage`)

Reply:
Curious: why is it found upon? Does it cause the dom diffing to be slower because it has to re-create the react element?

Reply:
I did a little digging and it seems like a debated topic actually - but at Wayfair the logic was, if you want to use a sub-render then that should probably be its own component entirely since you may be re-using it. If you are doing it to clean up a render function that that is probably also a sign that you should break up that component into smaller components. 

All render methods should be pretty easy to read without needing to follow a bunch of function calls to see what is being rendered.